### PR TITLE
Sponsor data access improvements

### PR DESCRIPTION
**Pull request by lalithr95**
This PR stores sponsors data in YAML file and loads file at application load time. Changes scope of sponsors method to private.

As initializer file is changed server need to be restarted to view updated changes.

### CODE REVIEW

File: app/models/sponsors.rb [hunk 1]
```diff
@@ -8,46 +8,11 @@ def find(identifier)
       OpenStruct.new(sponsors.fetch(identifier))
     end
 
+    private
+
+    # Fetches sponsors data at time of initalization. Sponsors_data is a global variable which stores sponsors
```
Comment (tgxworld) [inline app/models/sponsors.rb]: No need for this comment.

File: app/models/sponsors.rb [hunk 2]
```diff
@@ -8,46 +8,11 @@ def find(identifier)
...
-        bugsnag:
-        {
-          name: 'Bugsnag',
-          desc: "Detect and diagnose crashes in your applications.
-            Bugsnag's cross platform error monitoring helps you ship with confidence.",
-          link: 'https://bugsnag.com/',
-          class_name: 'sponsor-bugsnag'
-        }
-      }
+      Sponsors_data
```
Comment (tgxworld) [inline app/models/sponsors.rb]: In Ruby, the preference is to use CamelCase for classes.

File: config/initializers/load_settings.rb [hunk 3]
```diff
@@ -2,3 +2,6 @@
 
 settings = YAML.load_file(Rails.root.join('config', 'settings.yml'))[Rails.env]
 AppSettings = OpenStruct.new(settings) unless defined?(AppSettings)
+
+# Loads sponsor data from YAML file
```
Comment (tgxworld) [inline config/initializers/load_settings.rb]: No need for this comment.

File: config/initializers/load_settings.rb [hunk 4]
```diff
@@ -2,3 +2,6 @@
...
+::Sponsors_data = YAML.load_file(Rails.root.join('config', 'sponsors.yml'))
```
Comment (tgxworld) [inline config/initializers/load_settings.rb]: CamelCase here :)

File: test/test_helper.rb [hunk 5]
```diff
@@ -2,10 +2,12 @@
 require File.expand_path('../../config/environment', __FILE__)
 require 'rails/test_help'
 require 'mocha/mini_test'
+require 'capybara/rails'
```
Comment (tgxworld) [inline test/test_helper.rb]: This is already required in https://github.com/ruby-bench/ruby-bench-web/blob/master/test/acceptance/support/capybara.rb

File: test/test_helper.rb [hunk 6]
```diff
@@ -2,10 +2,12 @@
...
 
 Dir["#{Rails.root}/test/support/**/*"].each { |file| require file }
 
 class ActiveSupport::TestCase
   # Add more helper methods to be used by all tests here...
   include FactoryGirl::Syntax::Methods
+  include Capybara::DSL
```
Comment (tgxworld) [inline test/test_helper.rb]: We don't need `Capybara::DSL` for all tests. The module is only included for [accpetance tests](https://github.com/ruby-bench/ruby-bench-web/blob/1e3413a61eb5aa8823d7383425a9c10a84e81d47/test/acceptance/test_helper.rb)

File: test/integration/sponsors_test.rb [hunk 7]
```diff
@@ -0,0 +1,20 @@
+require 'test_helper'
+
+class SponsorsTest < ActionDispatch::IntegrationTest
+  test "should display same number of sponsors" do
+    visit root_path
+    click_link "Sponsors", match: :first
```
Comment (tgxworld) [inline test/integration/sponsors_test.rb]: It'll be good to use `I18n` here. https://github.com/ruby-bench/ruby-bench-web/blob/master/config/locales/en.yml#L74-L75

**Comment by tgxworld** [comment]
Hi @lalithr95! Thanks for the PR :+1:  I left some comments.

File: test/test_helper.rb (hunk 6 above)
Comment (lalithr95) [inline test/test_helper.rb]: @tgxworld  `Capybara::DSL` is required in this test, Shall I include only in this test ?

File: test/test_helper.rb (hunk 5 above)
Comment (lalithr95) [inline test/test_helper.rb]: @tgxworld  we require it in Integration tests(current test) as well, Shall I include it only with that test ?

Comment (tgxworld) [inline test/test_helper.rb]: Ah ic your confusion. The test should be moved under acceptance tests.

**Comment by lalithr95** [comment]
@tgxworld Updated code :+1:

File: app/models/sponsors.rb [hunk 8]
```diff
@@ -8,46 +8,10 @@ def find(identifier)
       OpenStruct.new(sponsors.fetch(identifier))
     end
 
+    private
+
     def sponsors
```
Comment (tgxworld) [inline app/models/sponsors.rb]: This method is not necessary since we have a Global constant now.

Comment (lalithr95) [inline app/models/sponsors.rb]: Ah I missed that. @tgxworld Updated have a look

**Comment by tgxworld** [comment]
Looks good. Thanks for the PR!

**Pull request merged**